server/app/mutations/stacks/create.rb

```diff
@@ -47,6 +61,26 @@ def execute
         return
       end
 
+      if self.volumes
+        # TODO Separate into own method
+        attributes.delete(:volumes)
+        self.volumes.each do |volume|
+          if volume[:external]
+            vol = Volume.where(name: volume['name'], grid: grid, stack: nil).first
+            if vol
+              stack.external_volumes.create!(volume: vol)
+            end
+          else
+            outcome = Volumes::Create.run(grid: grid, stack: stack, **volume.symbolize_keys)
+            unless outcome.success?
+              outcome.errors.message.each do |key, msg|
+                add_error(:volumes, :key, "Volume create failed for '#{volume[:name]}': #{msg}")
```

Comment:
This needs to also `return`, or the execute will just continue to create the stack and services.